packages/protobuf-test/src/message.test.ts

```diff
@@ -12,23 +12,155 @@
+      optionalStringPiece: "",
+      optionalCord: "",
+      repeatedInt32: [],
+      repeatedInt64: [],
+      repeatedUint32: [],
+      repeatedUint64: [],
+      repeatedSint32: [],
+      repeatedSint64: [],
+      repeatedFixed32: [],
+      repeatedFixed64: [],
+      repeatedSfixed32: [],
+      repeatedSfixed64: [],
+      repeatedFloat: [],
+      repeatedDouble: [],
+      repeatedBool: [],
+      repeatedString: [],
+      repeatedBytes: [],
+      repeatedNestedMessage: [],
+      repeatedForeignMessage: [],
+      repeatedNestedEnum: [],
+      repeatedForeignEnum: [],
+      repeatedStringPiece: [],
+      repeatedCord: [],
+      packedInt32: [],
+      packedInt64: [],
+      packedUint32: [],
+      packedUint64: [],
+      packedSint32: [],
+      packedSint64: [],
+      packedFixed32: [],
+      packedFixed64: [],
+      packedSfixed32: [],
+      packedSfixed64: [],
+      packedFloat: [],
+      packedDouble: [],
+      packedBool: [],
+      packedNestedEnum: [],
+      unpackedInt32: [],
+      unpackedInt64: [],
+      unpackedUint32: [],
+      unpackedUint64: [],
+      unpackedSint32: [],
+      unpackedSint64: [],
+      unpackedFixed32: [],
+      unpackedFixed64: [],
+      unpackedSfixed32: [],
+      unpackedSfixed64: [],
+      unpackedFloat: [],
+      unpackedDouble: [],
+      unpackedBool: [],
+      unpackedNestedEnum: [],
+      mapInt32Int32: {},
+      mapInt64Int64: {},
+      mapUint32Uint32: {},
+      mapUint64Uint64: {},
+      mapSint32Sint32: {},
+      mapSint64Sint64: {},
+      mapFixed32Fixed32: {},
+      mapFixed64Fixed64: {},
+      mapSfixed32Sfixed32: {},
+      mapSfixed64Sfixed64: {},
+      mapInt32Float: {},
+      mapInt32Double: {},
+      mapBoolBool: {},
+      mapStringString: {},
+      mapStringBytes: {},
+      mapStringNestedMessage: {},
+      mapStringForeignMessage: {},
+      mapStringNestedEnum: {},
+      mapStringForeignEnum: {},
+      oneofField: { case: undefined },
+      repeatedBoolWrapper: [],
+      repeatedInt32Wrapper: [],
+      repeatedInt64Wrapper: [],
+      repeatedUint32Wrapper: [],
+      repeatedUint64Wrapper: [],
+      repeatedFloatWrapper: [],
+      repeatedDoubleWrapper: [],
+      repeatedStringWrapper: [],
+      repeatedBytesWrapper: [],
+      optionalNullValue: NullValue.NULL_VALUE,
+      repeatedDuration: [],
+      repeatedTimestamp: [],
+      repeatedFieldmask: [],
+      repeatedStruct: [],
+      repeatedAny: [],
+      repeatedValue: [],
+      repeatedListValue: [],
+      fieldname1: 0,
+      fieldName2: 0,
+      FieldName3: 0,
+      fieldName4: 0,
+      field0name5: 0,
+      field0Name6: 0,
+      fieldName7: 0,
+      FieldName8: 0,
+      fieldName9: 0,
+      FieldName10: 0,
+      FIELDNAME11: 0,
+      FIELDName12: 0,
+      FieldName13: 0,
+      FieldName14: 0,
+      fieldName15: 0,
+      fieldName16: 0,
+      fieldName17: 0,
+      FieldName18: 0,
+    };
+
+    msg.optionalTimestamp = Timestamp.now();
```

Comment:
This assignment does not help us much. Consider the following example:

```ts
type A = {
  a: string;
}
type B = {
  a: string;
  b: string;
}
let a: A = {
  a: ""
}
let b: B = {
  a: "",
  b: ""
}
a = b;
```

`b` can be assigned to `a`, even though it has properties that `A` doesn't have. TypeScript simply does structural matching here, and only looks for the properties needed to satisfy `A`.

But consider this:

```ts
a = {
  a: "",
  b: "" // TS2322: Type '{ a: string; b: string; }' is not assignable to type 'A'.
        // Object literal may only specify known properties, and 'b' does not exist in type 'A'.
}
```

When assigning an object literal to a type, TypeScript _does_ check for excess properties. And we can make use of that fact to prove that a PlainMessage<Timestamp> field does not have any methods:

```suggestion
    msg.optionalTimestamp = {
      nanos: 0,
      seconds: protoInt64.zero
    };
```

Please add the same for the fields 
- `optionalDuration`
- `optionalAny`
- `optionalNestedMessage`

With that, we are starting to get some meaningful coverage. But PlainMessage is recursive now, which applies to repeated fields, maps, and oneofs. To cover those cases as well, please add assignments for the following fields:

- `repeatedDuration`
- `mapStringNestedMessage`
- `oneofField` for the case `"oneofNestedMessage"`

Reply:
Makes sense.  Thanks for the explanation.  Updated the PR.